src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatWriter.cs

```diff
@@ -11,112 +11,27 @@
+    internal class ReflectionXmlFormatWriter
     {
-        internal void ReflectionWriteClass(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext context, ClassDataContract classContract)
-        {
-            InvokeOnSerializing(obj, context, classContract);
-            obj = ResolveAdapterType(obj, classContract);
-            ReflectionWriteMembers(xmlWriter, obj, context, classContract, classContract, 0 /*childElementIndex*/);
-            InvokeOnSerialized(obj, context, classContract);
-        }
+        private ReflectionXmlClassWriter _reflectionClassWriter = new ReflectionXmlClassWriter();
 
-        private void InvokeOnSerializing(object obj, XmlObjectSerializerWriteContext context, ClassDataContract classContract)
+        public void ReflectionWriteClass(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext context, ClassDataContract classContract)
         {
-            if (classContract.BaseContract != null)
-                InvokeOnSerializing(obj, context, classContract.BaseContract);
-            if (classContract.OnSerializing != null)
-            {
-                var contextArg = context.GetStreamingContext() ;
-                classContract.OnSerializing.Invoke(obj, new object[] { contextArg });
-            }
+            _reflectionClassWriter.ReflectionWriteClass(xmlWriter, obj, context, classContract, null/*memberNames*/);
         }
 
-        private void InvokeOnSerialized(object obj, XmlObjectSerializerWriteContext context, ClassDataContract classContract)
+        public void ReflectionWriteCollection(XmlWriterDelegator xmlWriter, object obj, XmlObjectSerializerWriteContext context, CollectionDataContract collectionDataContract)
         {
-            if (classContract.BaseContract != null)
-                InvokeOnSerialized(obj, context, classContract.BaseContract);
-            if (classContract.OnSerialized != null)
-            {
-                var contextArg = context.GetStreamingContext() ;
-                classContract.OnSerialized.Invoke(obj, new object[] { contextArg });
-            }
-        }
+            XmlDictionaryString ns = collectionDataContract.Namespace;
+            XmlDictionaryString itemName = collectionDataContract.CollectionItemName;
 
-        private object ResolveAdapterType(object obj, ClassDataContract classContract)
-        {
-            Type type = obj.GetType();
-            if (type == Globals.TypeOfDateTimeOffset)
-            {
-                obj = DateTimeOffsetAdapter.GetDateTimeOffsetAdapter((DateTimeOffset)obj);
-            }
-            else if (type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == Globals.TypeOfKeyValuePair)
+            if (collectionDataContract.ChildElementNamespace != null)
             {
-                obj = classContract.KeyValuePairAdapterConstructorInfo.Invoke(new object[] { obj });
+                // TODO: should add tests for this.
```

Comment:
>               // TODO: should add tests for this. [](start = 1, length = 50)

What's the difficulty here? Do we not support this any more? Or is the implementation being deferred until there are tests to verify it? If so, open a bug and either remove the comment or reference the bug in the code.